### ISSUE TITLE
AudioWorkletRenderer output channel count iterator is flawed

### ISSUE THREAD

**Issue by CorvusPrudens** [orottier/web-audio-api-rs#413]
The iterator created over an AudioWorklet's output channels doesn't seem to do what its comment suggests.

https://github.com/orottier/web-audio-api-rs/blob/3a7687de542cc71a083f31078776552292702f5f/src/worklet.rs#L324-L330

There are three main cases here:
1. The worklet has one input and one output with the same amount of channels (and the output channel configuration vector was empty), and this iterator produces the correct result.
2. The worklet has a configuration other than one-input-one-output, in which case the iterator will always have one more output than it should, causing the worklet to panic later on.
3. On top of 2, if the worklet has no inputs, it will panic when the last element of the iterator is evaluated.

Something like this should work fine while avoiding heap allocation:
```rs
let single_case = [inputs.get(0).map(|i| i.number_of_channels()).unwrap_or_default()];

let output_channel_count = if self.output_channel_count.is_empty() {
    single_case.iter()
} else {
    self.output_channel_count.iter()
}.copied();
```

I'm sure there's a more elegant solution, though. I hope this helps!

**Comment by orottier** [comment]
Thanks for the report, you are right that the current implementation is incorrect. I'll put out a fix based on your suggestion and add some more tests.

This is a relatively new part of our library and has not been battle tested. You might encounter some more issues, please let us know if there's anything else that seems wrong or not intuitive.

**Comment by CorvusPrudens** [comment]
Oh awesome, I'm glad I could help! It's actually been very pleasant overall. The processor trait and construction process is super convenient for my project. I'll be sure to help with anything else I come across!

**Comment by orottier** [comment]
Fixed with #414 and released in 0.39.1
